google/cloud/bigtable/data/_metrics/handlers/opentelemetry.py

```diff
@@ -0,0 +1,196 @@
+        self.retry_count = meter.create_counter(
+            name="retry_count",
+            description="A count of additional RPCs sent after the initial attempt. Under normal circumstances, this will be 1.",
+        )
+        self.server_latencies = meter.create_histogram(
+            name="server_latencies",
+            description="A distribution of the latency measured between the time when Google's frontend receives an RPC and sending back the first byte of the response.",
+            unit="ms",
+        )
+        self.connectivity_error_count = meter.create_counter(
+            name="connectivity_error_count",
+            description="A count of the number of attempts that failed to reach Google's network.",
+        )
+        self.application_latencies = meter.create_histogram(
+            name="application_latencies",
+            description="A distribution of the total latency introduced by your application when Cloud Bigtable has available response data but your application has not consumed it.",
+            unit="ms",
+        )
+        self.throttling_latencies = meter.create_histogram(
+            name="throttling_latencies",
+            description="The latency introduced by the client by blocking on sending more requests to the server when there are too many pending requests in bulk operations.",
+            unit="ms",
+        )
+
+
+class OpenTelemetryMetricsHandler(MetricsHandler):
+    """
+    Maintains a set of OpenTelemetry metrics for the Bigtable client library,
+    and updates them with each completed operation and attempt.
+
+    The OpenTelemetry metrics that are tracked are as follows:
+      - operation_latencies: latency of each client method call, over all of it's attempts.
+      - first_response_latencies: latency of receiving the first row in a ReadRows operation.
+      - attempt_latencies: latency of each client attempt RPC.
+      - retry_count: Number of additional RPCs sent after the initial attempt.
+      - server_latencies: latency recorded on the server side for each attempt.
+      - connectivity_error_count: number of attempts that failed to reach Google's network.
+      - application_latencies: the time spent waiting for the application to process the next response.
+      - throttling_latencies: latency introduced by waiting when there are too many outstanding requests in a bulk operation.
+    """
+
+    def __init__(
+        self,
+        *,
+        project_id: str,
+        instance_id: str,
+        table_id: str,
+        app_profile_id: str | None = None,
+        client_uid: str | None = None,
+        instruments: _OpenTelemetryInstruments = _OpenTelemetryInstruments(),
+        **kwargs,
+    ):
+        super().__init__()
+        self.otel = instruments
+        # fixed labels sent with each metric update
+        self.shared_labels = {
+            "client_name": f"python-bigtable/{bigtable_version}",
+            "client_uid": client_uid or str(uuid4()),
```

Comment:
Is it possible to detect the hostname where the client is run similar to java? https://github.com/googleapis/java-bigtable/blob/main/google-cloud-bigtable-stats/src/main/java/com/google/cloud/bigtable/stats/BigtableStackdriverExportUtils.java#L158-L173 Or that's not possible with python?

Reply:
It should be possible to do something like that, I'll take a look

Reply:
done